### PR TITLE
Renderer: add Solid renderer

### PR DESCRIPTION
**Pull request by natemoo-re**
This PR adds a new renderer for [Solid](https://www.solidjs.com/)!

Warning: some messy branching here: This captures the work from #656 will be merged into #588, which will then be merged into `main`.

### CODE REVIEW

File: examples/framework-multiple/src/pages/index.astro [hunk 1]
```diff
@@ -32,6 +34,7 @@ import SvelteCounter from '../components/SvelteCounter.svelte';
             margin-bottom: 2em;
         }
     </style>
+    <HydrationScript />
```
Comment (eyelidlessness) [inline examples/framework-multiple/src/pages/index.astro:37]: I’m pretty sure this is unnecessary.

File: packages/renderers/renderer-solid/index.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  client: './client',
+  server: './server',
+  knownEntrypoints: ['solid-js', 'solid-js/web'],
+  external: ['solid-js/web/dist/server.js', 'solid-js/dist/server.js', 'babel-plugin-module-resolver', 'babel-preset-solid'],
+  jsxImportSource: 'solid-js',
+  jsxTransformOptions: async ({ isSSR }) => {
+    const [{ default: solid }] = await Promise.all([import('babel-preset-solid')]);
+    const options = {
+      presets: [
+        solid({}, { generate: isSSR ? 'ssr' : 'dom', hydratable: true }),
```
Comment (eyelidlessness) [inline packages/renderers/renderer-solid/index.js:12]: I'm also not sure you need `hydratable: true` here.

**Comment by praneybehl** [comment]
Wondering how far is it from getting merged?

**Comment by natemoo-re** [comment]
Not very far, I've just had a few other things come up. Getting back on top of this tomorrow!

**Comment by praneybehl** [comment]
Excellent, thank you 💯

File: examples/framework-multiple/src/pages/index.astro (hunk 1 above)
Comment (natemoo-re) [reply examples/framework-multiple/src/pages/index.astro:38]: Turns out it is! I was getting an error before with whatever method I was using.

**Comment by natemoo-re** [comment]
I'm going to merge this into #588! It will be merged into `main` from there after some documentation updates.

**Pull request merged**